purchase_vendor_promotion/tests/test_vendor_promotion.py

```diff
@@ -2,11 +2,14 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 from datetime import date
 
+from freezegun import freeze_time
+
 from odoo import Command
 from odoo.exceptions import ValidationError
 from odoo.tests import TransactionCase, tagged
 
 
+@freeze_time("2024-01-02 03:21:34")
```

Comment:
Decorating the class can give unexpected results, especially if a class is subclassed.
I always recommend to decorate the method or use the ctx manager where needed.

Reply:
As you a here... Can you use BaseCommon?

Reply:
> BaseCommon



> Decorating the class can give unexpected results, especially if a class is subclassed. I always recommend to decorate the method or use the ctx manager where needed.

I agree